manager/integration/tests/test_settings.py

```diff
@@ -278,47 +279,41 @@ def test_setting_toleration_extra(core_api, apps_api):  # NOQA
 def wait_for_toleration_update(core_api, apps_api, count,  # NOQA
                                expected_tolerations,
                                chk_removed_tolerations=[]):
-    not_managed_apps = [
-        "csi-attacher",
-        "csi-provisioner",
-        "csi-resizer",
-        "csi-snapshotter",
-        "longhorn-csi-plugin",
+    skip_updates = [
         "longhorn-driver-deployer",
         "longhorn-manager",
         "longhorn-ui",
     ]
-    updated = False
+    ok = False
     for _ in range(RETRY_COUNTS):
         time.sleep(RETRY_INTERVAL_LONG)
 
-        updated = True
+        ok = False
         if not check_workload_update(core_api, apps_api, count):
-            updated = False
             continue
 
         pod_list = core_api.list_namespaced_pod(LONGHORN_NAMESPACE).items
         for p in pod_list:
-            managed_by = p.metadata.labels.get('longhorn.io/managed-by', '')
-            if str(managed_by) != "longhorn-manager":
-                continue
-            else:
-                app_name = str(p.metadata.labels.get('app', ''))
-                assert app_name not in not_managed_apps
-
-            if p.status.phase != "Running" \
-                or not check_tolerations_set(p.spec.tolerations,
-                                             expected_tolerations,
-                                             chk_removed_tolerations):
-                updated = False
+            app_name = str(p.metadata.labels.get('app', ''))
+            if p.status.phase != "Running":
+                ok = False
```

Comment:
1. I am thinking if this pod-running check should be handled by `check_workload_update`
2. Some longhorn managed workloads like engine image pods do not contain the label `app`
3. Similar to `check_workload_update`, need to make sure all longhorn managed components only are listed and checked here

Reply:
Addressed.